pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1805,9 +1805,9 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 			},
 			validate: func(c client.Client, t *testing.T) {
 				cd := getCD(c)
-				// This is a hack to get Go to accept this arbitrary string as the proper type
-				dummy := hivev1.ClusterDeploymentCondition{Type: "ClusterImageSetNotFound"}
-				lc := controllerutils.FindCondition(cd.Status.Conditions, dummy.Type)
+				// This is a minor hack to get Go to accept this arbitrary string as the proper type
+				const clusterImageSetNotFound hivev1.ClusterInstallConditionType = "ClusterImageSetNotFound"
```

Comment:
I meant in apis/hive/v1/clusterinstall_conditions.go with the other `ClusterInstallConditionType` consts. Then there's no hackery -- this is just like any other `FindCondition` call.

Reply:
...and replace the hardcoded string on L325.

Perhaps this isn't in with the other consts because it's "legacy" and therefore we don't want consumers trying to look for it? I don't see why we can't have the const though -- just make sure the docstring comment mentions it's legacy and not to use it in consuming code.